api/server/acme/acme.go

```diff
@@ -0,0 +1,50 @@
+// Package acme abstracts away various ACME libraries
+package acme
+
+import (
+	"errors"
+	"net"
+
+	"golang.org/x/crypto/acme/autocert"
+)
+
+// ErrUnsupportedLibrary is returned by acme.New() if you
+// attempt to instantiate a library we don't support
+var ErrUnsupportedLibrary = errors.New("Unsupported Library")
+
+const (
+	// LibAutoCert is the acme library from golang.org/x/crypto/acme/autocert
+	LibAutoCert = "autocert"
+	// LibCertMagic is the acme library from github.com/mholt/certmagic
+	LibCertMagic = "certmagic"
+)
+
+// Library is an provider interface
+type Library interface {
+	NewListener([]string) (net.Listener, error)
```

Comment:
NewListener([]string) => NewListener(...string)

This would allow calling new listener without any hosts so it operates on any hostname. Also cleaner with call to NewListener() or NewListener("foo", "bar")